### PR TITLE
[WIP] Add Particle Argon, Boron and Xenon

### PR DESCRIPTION
**Pull request by suda**
This PR adds Particle Argon, Boron and Xenon boards to the docs.

It's blocked by https://github.com/tinygo-org/tinygo/pull/904 after which is merged, we can run `make doc-gen` to generate `machine` package docs for those boards.

### CODE REVIEW

**Comment by deadprogram** [comment]
Switched to `dev` branch so it gets picked up next release.

**Comment by conejoninja** [comment]
tinygo-org/tinygo#904 was merged, is anything besides the `make doc-gen` stopping this to be merged too?

**Comment by suda** [comment]
No, only `make doc-gen` is blocking. It's sill failing on `particle-argon` and I assumed it's because the https://github.com/tinygo-org/tinygo/pull/904 got merged into `dev` branch and doc-gen expects `master`. Is there a way to force generating docs from the dev branch?

**Comment by suda** [comment]
No, only `make doc-gen` is blocking. It's sill failing on `particle-argon` and I assumed it's because the https://github.com/tinygo-org/tinygo/pull/904 got merged into `dev` branch and doc-gen expects `master`. Is there a way to force generating docs from the dev branch?

**Comment by aykevl** [comment]
> Is there a way to force generating docs from the dev branch?

If you have the dev version of TinyGo installed (`go install`) it should just work.

**Comment by suda** [comment]
Oh installing made it run! It's just generating an empty file, i.e.:

```
---
title: particle-argon
---
```

It actually resets all files in `content/microcontrollers/machine`. Anything else I might be missing?

**Comment by aykevl** [comment]
I'm not sure what goes on there, it works for me (and includes the Particle boards).

**Comment by suda** [comment]
Ok, I got it working! Ready for review 😊

File: content/microcontrollers/particle-argon.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## Interfaces
+
+| Interface | Hardware Supported | TinyGo Support |
+| --------- | ------------- | ----- |
+| GPIO      | YES | YES |
+| UART      | YES | YES |
+| SPI      | YES | YES |
+| I2C      | YES | YES |
+| ADC      | YES | Not yet |
+| PWM      | YES | Not yet |
```
Comment (aykevl) [inline content/microcontrollers/particle-argon.md]: It looks like the ADC and PWM is supported? (Also goes for the other two boards)

Comment (suda) [reply content/microcontrollers/particle-argon.md]: I copied this from nRF52840-MDK. I wasn't sure if ADC and PWM are already implemented for nRF52. I'll try to test and confirm it works this weekend.

Comment (deadprogram) [reply content/microcontrollers/particle-argon.md]: Any action on this testing @suda ?

Comment (suda) [reply content/microcontrollers/particle-argon.md]: @deadprogram I did test both ADC and PWM just now and can confirm it works perfectly 👍 Pushed a commit updating this table

**Comment by deadprogram** [comment]
Now merging, thanks @suda

**Pull request merged**